packages/react-components/src/components/Survey/TagsSurvey/TagsSurvey.tsx

```diff
@@ -157,11 +157,18 @@ export const _TagsSurvey = (props: _TagsSurveyProps): JSX.Element => {
         if (issue) {
           setSelectedTags((prevState) => {
             if (prevState[issueCategory]?.issues) {
-              (prevState[issueCategory]!.issues as unknown[]) = prevState[issueCategory]!.issues!.filter(function (
-                value
-              ) {
-                return value !== issue;
-              });
+              // 'prevState[issueCategory]!.issues as ...' typing is required here to avoid a typescript limitation
+              // "This expression is not callable" caused by filter().
+              // More information can be found here: https://github.com/microsoft/TypeScript/issues/44373
+              (prevState[issueCategory]!.issues as (_AudioIssue | _OverallIssue | _ScreenshareIssue | _VideoIssue)[]) =
```

Comment:
Can this be cleaned up at all? 
Ultimately, the filter is is if the value is equal to the issue - is there a way to make this simpler?

Reply:
This is not an easy unfortunately. More info can be found in this github typescript issue if you would like more details regarding it https://github.com/microsoft/TypeScript/issues/36390.

One way to fix it is through typing arrays like this:
something: (object | object)[]. 
But in this case, prevState is already a _CallSurvey type. No further typing can be enhanced here. If you have any other suggestions please let me know, I'll be more than happy to try it out.

Reply:
Ok, that's a fun long-running TS issue....

Reply:
Out of interest, how do this only appear in this PR and not in the react-components PR?

Reply:
Oh it was previously `as unknown`, this is just slightly stronger - nice!

Reply:
In setStates, it is not best practice to mutate the prevState and return the mutated state. You are better to create a new object and return that.

I'm struggling to fully understand this snippet, but is this the same as:
```ts
setSelectedTags((prevState) => {
  const categoryIssues = prevState[issueCategory]?.issues?.filter((value) => value !== issue);
  return {
    ...prevState,
    [issueCategory]: { issues: categoryIssues }
  };
});
```
? If it is that would simplify signifanctly

Reply:
But honestly I find this whole code block very hard to parse so that could be wrong

Reply:
With this snippet code we still hit the "filter" issue as mentioned in the previous comments. So an as typing is still required. I'll continue to do more testing with surveyTags to see if the snippet code causes any changes to the functionality.

Reply:
Tested on Calling sample and is functioning correctly :D